### DISCUSSION TITLE
iOS native code of get alarm audio

### DISCUSSION THREAD

**Discussion by whowillcare** [gdelataillade/alarm#41] [Ideas]
Will you consider adding code like the following code to make audio player to be play asset audio defined in flutter instead of as you instructed:
To import your alarm audio(s), you will need to drag and drop your asset(s) to your Runner folder in Xcode.

 (I personally didn't test this method at all, but in theory, it should work, please let me know if this code actually helps or not, thanks)

```swift
...
public class SwiftAlarmPlugin: NSObject, FlutterPlugin {
...
 public static func register(with registrar: FlutterPluginRegistrar) {
    ...
    instance.registrar = registrar
    ...
  }
...
private var registrar: FlutterPluginRegistrar!
...
  private func setAlarm(call: FlutterMethodCall, result: FlutterResult) {
     ...
     if let audioPath = Bundle.main.path(forResource: self.registrar.lookupKey(forAsset: assetAudio), ofType: nil) {
  ...
}
```

**Comment by gdelataillade** [comment]
Hi @whowillcare 

I never tried to do this way. It looks like it worth giving it a shot !

Thanks for your time.

**Comment by gdelataillade** [comment]
Hey @whowillcare,

I found the time to take a look at your suggestion and it works great ! I've added it to `1.1.4`.

Thank you again.

**Comment by whowillcare** [comment]
That's great news.
